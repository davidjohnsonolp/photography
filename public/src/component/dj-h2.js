import buildShadowRoot from '../lib/build-shadow-root.js';

class DJH2 extends HTMLElement {
  constructor() {
    super();
    const html = `
      <style>
        :host {
        }
        h2 {
          display: block;
          font-size: 1.5em;
          margin: 0.83em 0;
          font-weight: bold;
        }
      </style>
      <h2>
        <slot></slot>
      </h2>
    `;
    buildShadowRoot(html,this);
  }
}

customElements.define('dj-h2', DJH2);
export default DJH2;
