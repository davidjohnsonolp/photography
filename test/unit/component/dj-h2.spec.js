import '../../../public/src/component/dj-h2.js';

let customH2;

describe('DJH2 Custom H2', () => {
  beforeEach(() => {
    customH2 = window.document.createElement('dj-h2'); 
    document.body.appendChild(customH2);
  });

  afterEach(() => {
    document.body.getElementsByTagName('dj-h2')[0].remove();
  });

  it('renders correctly', () => {
    expect({ html: customH2.shadowRoot.innerHTML }).toMatchSnapshot();
  });
});