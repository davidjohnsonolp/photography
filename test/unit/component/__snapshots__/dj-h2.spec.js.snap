// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DJH2 Custom H2 renders correctly 1`] = `
Object {
  "html": "
      <style>
        :host {
        }
        h2 {
          display: block;
          font-size: 1.5em;
          margin: 0.83em 0;
          font-weight: bold;
        }
      </style>
      <h2>
        <slot></slot>
      </h2>
    ",
}
`;
